src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php

```diff
@@ -311,10 +311,16 @@ private function detectMappingType(string $directory, ContainerBuilder $containe
         foreach ($glob as $file) {
             $content = file_get_contents($file);
 
-            if (preg_match('/^#\[.*'.$quotedMappingObjectName.'\b/m', $content)) {
+            if (
+                preg_match('/^#\[.*'.$quotedMappingObjectName.'\b/m', $content) ||
+                preg_match('/^#\[.*'.preg_quote('Embeddable', '/').'\b/m', $content)
```

Comment:
Remove the preg_quote call entirely here (thus, for this particular case, it is a no-op as there is no special characters). And even if there were some, it would still be a good idea to apply the necessary escaping in the source code to keep a literal regexp when it does not need to be dynamic.